plugin/storage/es/samplingstore/storage.go

```
@@ -0,0 +1,189 @@
+	s.writeThroughput(writeIndexName, ts, throughput)
+	return nil
+}
+
+func (s *SamplingStore) InsertProbabilitiesAndQPS(hostname string,
+	probabilities model.ServiceOperationProbabilities,
+	qps model.ServiceOperationQPS,
+) error {
+	ts := time.Now()
+	writeIndexName := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, ts)
+	val := dbmodel.ProbabilitiesAndQPS{
+		Hostname:      hostname,
+		Probabilities: probabilities,
+		QPS:           qps,
+	}
+	s.writeProbabilitiesAndQPS(writeIndexName, ts, val)
+	return nil
+}
+
+func (s *SamplingStore) writeProbabilitiesAndQPS(indexName string, ts time.Time, pandqps dbmodel.ProbabilitiesAndQPS) {
+	s.client().Index().Index(indexName).Type(probabilitiesType).
+		BodyJson(&dbmodel.TimeProbabilitiesAndQPS{
+			Timestamp:           ts,
+			ProbabilitiesAndQPS: pandqps,
+		}).Add()
+}
+
+func indexWithDate(indexNamePrefix, indexDateLayout string, date time.Time) string {
+	return indexNamePrefix + date.UTC().Format(indexDateLayout)
+}
+
+func (s *SamplingStore) writeThroughput(indexName string, ts time.Time, throughputs []*model.Throughput) {
+	s.client().Index().Index(indexName).Type(throughputType).
+		BodyJson(&dbmodel.TimeThroughput{
+			Timestamp:  ts,
+			Throughput: dbmodel.FromThroughputs(throughputs),
+		}).Add()
+}
+
+func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput, error) {
+	ctx := context.Background()
+	indices := s.getReadIndices(start, end)
+	searchResult, err := s.client().Search(indices...).
+		Size(s.maxDocCount).
+		Query(buildTSQuery(start, end)).
+		IgnoreUnavailable(true).
+		Do(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to search for throughputs: %w", err)
+	}
+	var retSamples []dbmodel.Throughput
+	hits := searchResult.Hits.Hits
+	for _, hit := range hits {
+		source := hit.Source
+		var tToD dbmodel.TimeThroughput
+		if err := json.Unmarshal(*source, &tToD); err != nil {
+			return nil, errors.New("unmarshalling ElasticSearch documents failed")
```

don't swallow errors
```suggestion
			return nil, errors.New("unmarshalling ElasticSearch documents failed: %w", err)
```

---

don't need ElasticSearch in the message